src/iminuit/cost.py

```diff
@@ -676,6 +787,77 @@ def __init__(self, n, xe, model, verbose):
         super().__init__(describe(model)[1:], self._prepare_data(n), verbose)
 
 
+class BarlowBeestonLite(MaskedCost):
+    """
+    Binned cost function for a template fit with uncertainties on the template.
+
+    Compared to the Beeston-Barlow method, the lite method uses one nuisance
+    parameter per bin instead of one nuisance parameter per component per bin,
+    and replaces the Poisson constraint with a Gaussian on the multiplicative
+    nuisance parameter.
+
+    J.S. Conway, PHYSTAT 2011, https://doi.org/10.48550/arXiv.1103.0354
+    P. Mandrik, https://arxiv.org/abs/1708.07708
+    """
+
+    __slots__ = ("_templates", "_sum_templates")
+
+    def __init__(
+        self, n, templates, name: _tp.Collection[str] = None, verbose: int = 0
+    ):
+        """
+        Initialize cost function with data and model.
+
+        Parameters
+        ----------
+        n : array-like
+            Histogram counts.
+        templates : array-like
+            TODO
+        name : collection of str, optional
+            Parameter names.
+        verbose : int, optional
+            Verbosity level. 0: is no output (default).
+            1: print current args and negative log-likelihood value.
+        """
+        n = np.atleast_1d(n)
+        templates = np.atleast_2d(templates)
+        M = len(templates)
+        if M < 1:
+            raise ValueError("at least one template is required")
+        if n.shape != templates.shape[1:]:
+            raise ValueError("shapes of n and templates do not match")
+        self._templates = templates
+        self._sum_templates = np.sum(templates, axis=1)
+        if name is None:
+            name = [f"x{i}" for i in range(M)]
+        else:
+            if len(name) != M:
+                raise ValueError("number of names must match number of templates")
+        super().__init__(name, n, verbose)
+
+    @Cost.ndata.getter  # type:ignore
+    def ndata(self):
+        """See Cost.ndata."""
+        shape = self._masked.shape
+        return np.prod(shape)
+
+    def _call(self, args):
+        mu = 0
+        mu_var = 0
+        for a, t, nt in zip(args, self._templates, self._sum_templates):
```

Comment:
In continuation of my previous comment on line 830
I would use numpy arrays here instead of for loops:
```python
#might need to add a new axis to enable multiplication to work properly 
#args = np.asarray(args) 
args = np.asarray(args) [:, np.newaxis]
mu = args * self.__templates
mu_var = self.__variances * args**2
```
I think if you provide the templates normalised to the data as in the previous comment (line 830), you would not need to normalise the fit fractions/args with respect to the templates anymore.
And with numpy arrays you get faster calculations + it's easier to understand

Reply:
I understand your point, but I intentionally do use that here. It is not significantly faster in this case and `args = np.asarray(args) [:, np.newaxis]` is not beautiful code.

Reply:
fair enough